### PR TITLE
Only Bitcoin .IsBitcoin()

### PR DESCRIPTION
**Pull request by codingupastorm**
Just because it's not Stratis, doesn't mean it's Bitcoin ;)

A big thing to beware of is that anyone running a BTC node will need to rename their node storage folder. I'm hoping this is acceptable for now... (noone's running a Bitcoin node are they?)

### CODE REVIEW

**Comment by codingupastorm** [comment]
Oops, let me fix the failing tests. Still worth reviewing though

File: src/Stratis.CirrusD/Properties/launchSettings.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (bokobza) [inline src/Stratis.CirrusD/Properties/launchSettings.json]: This thing's crept in.

File: src/Stratis.Bitcoin.Tests/Utilities/NetworkExtensionsTest.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            Assert.False(Networks.Networks.Stratis.Mainnet().IsBitcoin());
+            Assert.False(Networks.Networks.Stratis.Testnet().IsBitcoin());
+            Assert.False(Networks.Networks.Stratis.Regtest().IsBitcoin());
+
+            Assert.False(new ImaginaryPoANetwork().IsBitcoin());
+            Assert.False(new ImaginaryCirrusNetwork().IsBitcoin());
+        }
+
+        private class ImaginaryPoANetwork : Network
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Tests/Utilities/NetworkExtensionsTest.cs]: lol

**Comment by dangershony** [comment]
Looks good to me, fix tests and we push

**Comment by quantumagi** [comment]
`IsBitcoin` is used in 3 places:

image

image

image

I would suggest that we do away with using properties named `IsBitcoin`, `IsStratis` or even `IsCirrus`. Instead we should explicitly specify what should be different for each network. That can be achieved by extending the command line arguments using a property defined on the `Network` class - e.g. `network.args = new [] {"-useLegacyExtPubKeyConverter=1`", "-defaultApiPort=37221", "-useLightWalletBitcoinExternalFeePolicy=1"}`.

**Comment by codingupastorm** [comment]
@quantumagi - I _really_ like this idea! I actually need to use the IsBitcoin in a similar way to how it's being used above... for default params.

I see @mikedennis did a thumbs up.

@dangershony @fassadlr can I build this? A `defaultArgs` object on Network? 

Only question I have is where would the args come in the config hierarchy?

I guess it's Cmd line > defaultArgs > Config file ?

**Comment by quantumagi** [comment]
config file, defaults to, and also overrides defaultArgs. Cmd line overrides Config file.

Internal default constants are overridden by defaultArgs.

**Comment by codingupastorm** [comment]
Okay, I still think that the current method IsBitcoin() needs to be fixed, so it would be good if it goes in, and overhauling config is a separate issue.

**Comment by quantumagi** [comment]
I agree. Just a final note though; A conditional statement based on a variable such as Bitcoin which made sense when there was only 2 networks can't simply be updated for multiple networks without reviewing what the conditional statement set out to achieve originally. Just be sure that the `else` case caters to all other networks... Otherwise it would really be safer to do away with the if/else logic and the corresponding variable that drives it.

**Comment by dangershony** [comment]
I suggest we actually delete all the `IsBitcoin` occurrences (and IsStratis) and fix it by design

File: src/Stratis.Bitcoin.Tests/Utilities/NetworkExtensionsTest.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+namespace Stratis.Bitcoin.Tests.Utilities
+{
+    public class NetworkExtensionsTest
+    {
+        [Fact]
+        public void BitcoinNetworksAreBitcoin()
+        {
+            Assert.True(Networks.Networks.Bitcoin.Mainnet().IsBitcoin());
+            Assert.True(Networks.Networks.Bitcoin.Mainnet().IsBitcoin());
+            Assert.True(Networks.Networks.Bitcoin.Mainnet().IsBitcoin());
```
Comment (MithrilMan) [inline src/Stratis.Bitcoin.Tests/Utilities/NetworkExtensionsTest.cs]: I suppose these 3 lines are wrong, you want to test against Mainnet,testnet and regtest

**Comment by codingupastorm** [comment]
Going to close this until we know the direction we want to take here.

**Pull request closed without merging**